### PR TITLE
WPC-212 Rich Text in display only mode

### PR DESCRIPTION
**Pull request by bishoymelek**
- Editing is disabled.

- The toolbar is hidden.

### CODE REVIEW

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 1]
```diff
@@ -16,13 +16,16 @@ import {LinkFormatButton, RemoveLinkFormatButton} from './toolbar/linkButton'
 import {TableMenu} from './toolbar/tableMenu'
 import {WepublishEditor} from './editor/wepublishEditor'
 
-export type RichTextBlockProps = BlockProps<RichTextBlockValue>
+export interface RichTextBlockProps extends BlockProps<RichTextBlockValue> {
+  displayOnly: boolean
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:20]: make it optional

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 2]
```diff
@@ -16,13 +16,16 @@ import {LinkFormatButton, RemoveLinkFormatButton} from './toolbar/linkButton'
...
+}
 
 export const RichTextBlock = memo(function RichTextBlock({
   value,
   autofocus,
   disabled,
-  onChange
+  onChange,
+  displayOnly
```
Comment (roosnic1) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:28]: add a default value `displayOnly = false`

**Review by roosnic1 — CHANGES_REQUESTED** [review]

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx (hunk 2 above)
Comment (faebebin) [reply packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:28]: Do we need this additional arg? Couldn't just say if `disabled` then hide toolbar?

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx [hunk 3]
```diff
@@ -54,60 +57,62 @@ export const RichTextBlock = memo(function RichTextBlock({
...
+          <SubMenuButton icon="smile-o">
+            <EmojiPicker setEmoji={emoji => editor.insertText(emoji)} />
+          </SubMenuButton>
+        </Toolbar>
+      )}
       {WepublishEditor.isEmpty(editor) && ( // Alternative placeholder
         <div onClick={() => ReactEditor.focus(editor)} style={{color: '#cad5e4'}}>
           {t('blocks.richText.startWriting')}
         </div>
       )}
```
Comment (faebebin) [inline packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:113]: Think the conditional hiding should include the placeholder, as it states 'start writing' ...

**Review by faebebin — CHANGES_REQUESTED** [review]
Do we need `disabled` and `displayOnly` - isn't it same?

**Comment by roosnic1** [comment]
@faebebin i think so. Sometimes (e.g.: connetion lost) we just want to disable the editor and other times we just want to display RichText.

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx (hunk 2 above)
Comment (bishoymelek) [reply packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:28]: @roosnic1 why add default value false?

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx (hunk 3 above)
Comment (bishoymelek) [reply packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:115]: Fixed :ok_hand:

File: packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx (hunk 2 above)
Comment (roosnic1) [reply packages/editor/src/client/blocks/richTextBlock/richTextBlock.tsx:28]: @bishoymelek because `displayOnly` is a boolean and shouldn't be `undefined`

**Review by roosnic1 — CHANGES_REQUESTED** [review]

**Review by faebebin — DISMISSED** [review]

**Review by HanaAyman — APPROVED** [review]

**Review by roosnic1 — APPROVED** [review]

**Review by HanaAyman — APPROVED** [review]

**Pull request merged**